neo/rawio/spikegadgetsrawio.py

```diff
@@ -0,0 +1,247 @@
+
+class SpikeGadgetsRawIO(BaseRawIO):
+    extensions = ['rec']
+    rawmode = 'one-file'
+
+    def __init__(self, filename='', selected_streams=None):
+        """
+        
+        filename: str
+            filename ".rec"
+        
+        streams: None, list, str
+            sublist of streams to load/expose to API
+            uselfull for spikeextractor when one stream isneed.
+            For instance streams = ['ECU', 'trodes']
+            'trodes' is name for ephy channel (ntrodes)
+        """
+        BaseRawIO.__init__(self)
+        self.filename = filename
+        self.selected_streams = selected_streams
+
+    def _source_name(self):
+        return self.filename
+
+    def _parse_header(self):
+        
+        # parse file until "</Configuration>"
+        header_size = None
+        with open(self.filename, mode='rb') as f:
+            while True:
+                line = f.readline()
+                if b"</Configuration>" in line:
+                    header_size = f.tell()
+                    break
+
+            if header_size is None:
+                ValueError("SpikeGadgets : the xml header do not contain </Configuration>")
+            
+            f.seek(0)
+            header_txt = f.read(header_size).decode('utf8')
+            
+        # explore xml header
+        root = ElementTree.fromstring(header_txt)
+        gconf = sr = root.find('GlobalConfiguration')
+        hconf = root.find('HardwareConfiguration')
+        sconf = root.find('SpikeConfiguration')
+        
+        self._sampling_rate = float(hconf.attrib['samplingRate'])
+        num_ephy_channels = int(hconf .attrib['numChannels'])
+        
+        # explore sub stream and count packet size
+        # first bytes is 0x55
+        packet_size = 1
+        stream_bytes = {}
+        for device in hconf:
+            stream_id = device.attrib['name']
+            num_bytes = int(device.attrib['numBytes'])
+            stream_bytes[stream_id] = packet_size
+            packet_size += num_bytes
+        
+        # timesteamps 4 uint32
+        self._timestamp_byte = packet_size
+        packet_size += 4
+        
+        packet_size += 2 * num_ephy_channels
+
+        # read the binary part lazily
+        raw_memmap = np.memmap(self.filename, mode='r', offset=header_size, dtype='<u1')
+
+        num_packet = raw_memmap.size // packet_size
+        raw_memmap = raw_memmap[:num_packet*packet_size]
+        self._raw_memmap = raw_memmap.reshape(-1, packet_size)
+
+        # create signal channels
+        stream_ids = []
+        signal_streams = []
+        signal_channels = []
+        
+        # walk in xml device and keep only "analog" one
+        self._mask_channels_bytes  = {}
+        for device in hconf:
+            stream_id = device.attrib['name']
+            for channel in device:
+
+                if 'interleavedDataIDByte' in channel.attrib:
+                    # TODO deal with "headstageSensor" wich have interleaved
```

Comment:
which have interleaved what?
```suggestion
                    # TODO deal with "headstageSensor" which have interleaved
```